base/abstractarraymath.jl

```diff
@@ -88,12 +88,18 @@ function _dropdims(A::AbstractArray, dims::Dims)
             dims[j] == dims[i] && throw(ArgumentError("dropped dims must be unique"))
         end
     end
+    ox = axes(A)
     ax = _foldoneto((ds, d) -> d in dims ? ds : (ds..., axes(A,d)), (), Val(ndims(A)))
-    reshape(A, ax::typeof(_sub(axes(A), dims)))
+    if isconcretetype(eltype(ox))
+        # if all the axes are the same type, we can use the tail as the
+        # axes of the result rather than extracting one at each index
+        return reshape(A, ax::typeof(_sub(ox, dims)))
+    else
+        return reshape(A, ax)
+    end
```

Comment:
I'd deduplicate this by moving the `reshape` call out of the branch:

```suggestion
    if isconcretetype(eltype(ox))
        # if all the axes are the same type, we can use the tail as the
        # axes of the result rather than extracting one at each index
        ax = ax::typeof(_sub(ox, dims))
    end
    return reshape(A, ax)
```

Reply:
I actually do prefer to have the return values on the same indentation. I suppose we could golf one word by making it a `return if`, but I don't know if that's really worth the CI churn